### ISSUE TITLE
px-column with type=html need string of html to render html

### ISSUE THREAD

**Issue by LRagji** [predixdesignsystem/px-data-table#188]
##Looking for recommended Pattern

Currently when i declare a column with type html i am forced to have the html as string for that data item bound which is then rendered on the screen is this the only pattern? the problem with this seems to me is we are now injecting view in the view-model

## Code example
Linking an existing code pen which i found in bugs section:https://codepen.io/randyaskin/pen/apexgB?editors=1010

**Comment by benmak11** [comment]
@LRagji I believe this [documentation](https://www.predix-ui.com/#/elements/data-table/px-data-table-column) covers the px-data-column data types.

It might not be the only way to render on screen but if you have suggestions I would love to hear

**Comment by LRagji** [comment]
I was thinking can it be more declarative something like this(template column)
`<px-data-table-column name="status" type="html">
       <px-alert-label type=warning label={{row.item}}></px-alert-label"}
</px-data-table-column>`
Problem is i have to save the html back into my model or inject it into the model at the view model side
Second issue i found was while filtering on this column all i get is the html text not even the dom of the string created html...

**Comment by randyaskin** [comment]
The forthcoming Vaadin-based version of px-data-table will support templating of columns instead of requiring you to pass them as strings in your data model.

**Comment by Zynde** [comment]
@randyaskin I only recently noticed a couple of things that lead me to think something was happening with Predix and Vaadin. I am using Vaadin 10 (Flow) in a new project and want to make use of Predix components. I was looking at upgrading some to Polymer 2 which seems to have taken place and I was looking at trying to get the Vaadin 10 hookup done. Since I noticed your comment about Vaadin can you please tell me if you are doing a Vaadin 10 version of all the Predix components? I also saw a comment elsewhere that you may have worked on the code already does that mean the work is some way to having them ready soon if indeed you are doing that?

Any information greatly appreciated and would reduce my workload to do the same.

**Comment by randyaskin** [comment]
@Zynde we have contracted Vaadin to provide us a Predix-themed wrapper around their vaadin-grid element, as it already has many of the features that have been requested in px-data-table. The new component is written in Polymer 2 to go along with all of our other Predix components that are now upgraded to 1.x/2.x hybrid. We expect to have delivery of the new grid by the end of the year, though obviously with the US holidays next week we may need a bit more time to fully test and certify it.

I'm not familiar with Vaadin Flow / Vaadin 10, although I looked up an overview on their website. We certainly haven't talked to Vaadin about anything Java-related, and I have no idea if our components would now work with that product since they're now compatible with Polymer 2. ¯\_(ツ)_/¯ Let us know what you find out!
